File: src/main/java/org/jukeboxmc/potion/SlownessEffect.java
package org.jukeboxmc.potion;

import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.entity.EntityLiving;

import java.awt.*;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class SlownessEffect extends Effect {

    @Override
    public int getId() {
        return 2;
    }

    @Override
    public @NotNull EffectType getEffectType() {
        return EffectType.SLOWNESS;
    }

    @Override
    public @NotNull Color getEffectColor() {
        return new Color( 90, 108, 129 );
    }

    @Override
    public void update( long currentTick ) {

    }

    @Override
    public void apply(@NotNull EntityLiving entityLiving ) {
        entityLiving.setMovement(entityLiving.getMovement() * (1 - 0.15f * (this.amplifier + 1)));
    }

    @Override
    public void remove(@NotNull EntityLiving entityLiving ) {
        entityLiving.setMovement(entityLiving.getMovement() / (1 - 0.15f * (this.amplifier + 1)));
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/effect/SlownessEffect.kt
package org.jukeboxmc.server.effect

import org.jukeboxmc.api.effect.EffectType
import org.jukeboxmc.server.entity.JukeboxEntityLiving
import java.awt.Color

class SlownessEffect : JukeboxEffect() {

    override fun getId(): Int {
        return 2
    }

    override fun getEffectType(): EffectType {
        return EffectType.SLOWNESS
    }

    override fun getEffectColor(): Color {
        return Color(90, 108, 129)
    }

    override fun apply(entityLiving: JukeboxEntityLiving) {
        entityLiving.setMovement(entityLiving.getMovement() * (1 - 0.15f * (this.getAmplifier() + 1)))
    }

    override fun update(currentTick: Long) {

    }

    override fun remove(entityLiving: JukeboxEntityLiving) {
        entityLiving.setMovement(entityLiving.getMovement() / (1 - 0.15f * (this.getAmplifier() + 1)))
    }
}